### DISCUSSION TITLE
Como funcionan las Preguntas Dependientes en las Encuestas

### DISCUSSION THREAD

**Discussion by ingenierocvasquez** [chamilo/chamilo-lms#5161] [General] [answered]
Muy buenos días, cordial saludo, tengo una duda con respecto al siguiente código de configuración en las encuestas donde puedo evidenciar preguntas dependientes según sea la opción de respuesta.

// Survey
// ALTER TABLE c_survey_question ADD parent_id INT(11) DEFAULT 0 NOT NULL;
// ALTER TABLE c_survey_question ADD parent_option_id INT(11) DEFAULT 0 NOT NULL;
$_configuration['survey_question_dependency'] = true;

He buscado información al respecto, pero no he encontrado documentación al respecto.

Muchas gracias.

**Comment by ywarnier** [comment, marked as answer]
Hola @ingenierocvasquez 

Tienes razón, se nos pasó documentar esta funcionalidad. La idea es que puedas definir que una pregunta solo *aparece* si es que se ha seleccionado una cierta respuesta de una pregunta "pariente" (solo en modo radio-button para la pregunta pariente).

Para esto, tienes que ejecutar los dos comandos "ALTER" en tu base de datos, luego activar survey_question_dependency como lo hiciste, y luego al crear una encuesta tiene que haber creado una pregunta principal que servirá de "madre", y luego crear la(s) pregunta(s) hija(s). Al momento de crear cualquier pregunta, ahora te aparecerán un campo "pariente" y un campo "opción" que permiten seleccionar la pregunta pariente y la única opción de esta pregunta pariente que desata el gatillo para que aparezca tu pregunta hija.

Aquí te pongo una captura (en francés, pero se entiende facilmente) de como se ve al editar la hija.

![example-question](https://github.com/chamilo/chamilo-lms/assets/609839/bcf7160a-8b3b-4c43-9554-2c9215e9484d)

**Comment by ingenierocvasquez** [comment]
Muchas gracias por la Respuesta.
